src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java

```diff
@@ -3,37 +3,63 @@
 import java.util.Objects;
 
 public class TimeSlot {
-    
+
     private Day day;
-    private int hour;
-    
-    // TODO: is this clear?
-    private int duration;
-    
-    public TimeSlot(Day day, int hour, int duration) {
+    private static final int MIN_HOUR = 8;
+    private static final int MAX_HOUR = 20;
+    private int startHour = MIN_HOUR;
+    private int endHour = MAX_HOUR;
+
+    public TimeSlot(Day day, int startHour, int endHour) throws IllegalArgumentException {
         this.day = day;
-        this.hour = hour;
-        this.duration = duration;
+
+        if (isStartHourCorrect(startHour)) {
+            this.startHour = startHour;
+        } else {
+            throw new IllegalArgumentException("Argument startHour is not correct!");
+        }
+
+        if (isEndHourCorrect(endHour)) {
+            this.endHour = endHour;
+        } else {
+            throw new IllegalArgumentException("Argument endHour is not correct!");
+        }
     }
 
     public Day getDay() {
         return day;
     }
 
-    public int getHour() {
-        return hour;
+    public int getStartHour() {
+        return startHour;
+    }
+
+    public int getEndHour() {
+        return endHour;
+    }
+
+    private boolean isStartHourCorrect(int startHour) {
+        return startHour < endHour && startHour >= MIN_HOUR && startHour < MAX_HOUR;
```

Comment:
I think that these checks are quite confusing... You're doing check using parameter startHour and also attribute endHour. You're mixing parameters and attributes and method name doesn't imply anything like this. Check for both parameters maybe would be better